ttnn/CMakeLists.txt

```
@@ -1,9 +1,166 @@
+####################################################################################################
+# Build TTNN and TTNNCPP
+####################################################################################################
+
+##################################################
+# Set options
+##################################################
 if(ENABLE_COVERAGE AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     message(STATUS "Enabling code coverage flags for all ttnn targets")
     add_compile_options(--coverage)
     add_link_options(--coverage)
 endif()
 
+##################################################
+# Define a method build_library to define all the options and properties for a library
+##################################################
+function(
+    build_library
+    LIBRARY_NAME
+    SOURCES
+    LINK_DIRECTORIES
+    INCLUDE_DIRS
+    TARGET_LINK_LIBRARIES
+)
+    ##################################################
+    # Declare and define the library
+    ##################################################
+
+    #todo: Ideally declaring the library should be the first thing to do, sadly making it shared requires to add sources
+    string(TOUPPER "${LIBRARY_NAME}" LIBRARY_NAME_ALIAS)
+    add_library(${LIBRARY_NAME} SHARED ${SOURCES})
+    add_library(Metalium::${LIBRARY_NAME_ALIAS} ALIAS ${LIBRARY_NAME})
```

Metal == TT-Metalium (but the dir hasn't been renamed yet).  TT-NN builds on top of Metalium.  We're just currently sharing a repo.  The repo name would be more accurately named "TT-Metalium, et al" (we also have TT-Train in here, too).

```suggestion
    add_library(TT::NN::${LIBRARY_NAME_ALIAS} ALIAS ${LIBRARY_NAME})
```